Dockerfile

```
@@ -8,8 +8,21 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 COPY requirements.txt .
-RUN pip install --no-cache-dir -r requirements.txt
-RUN pip install -U "ell-ai[all]"
+
+# Upgrade pip and configure for better network resilience
+RUN pip install --upgrade pip setuptools wheel
+
+# Install large scientific packages first with retries and longer timeout
+RUN pip install --no-cache-dir --timeout=300 --retries=5 \
```

Hadolint (2.14.0)

[error] Multiple redirections compete for stdout. Use cat, tee, or pass filenames instead.

(SC2261)